### ISSUE TITLE
[question / proposal] Continuous integration

### ISSUE THREAD

**Issue by extensia** [jrsoftware/issrc#174]
Has anyone considered setting up continuous integration / release automation for the Inno Setup projects?

There are some logistical challenges such as access to old Delphi versions, but it would not be overly difficult to get up and running. We do this internally for dozens of libraries and applications and would be happy to help out.

**Comment by martijnlaan** [comment]
I use Jenkins myself for work stuff but not for Inno Setup. For Inno Setup a build.bat file is included which builds everything.

**Comment by DeeHants** [comment]
Questions like this should really be aimed at the [newsgroups](http://www.jrsoftware.org/newsgroups.php).
Issues are for specific bugs or well defined feature requests (which should really be discussed on the aforementioned newsgroups first)

**Comment by extensia** [comment]
Sorry. I have not used the forums in years and assumed things had moved to github infrastructure.

I brought up CI for Inno Setup since it is not the easiest application to build given the ancient compiler versions used. I have compilers going back to TP3, but I suspect most people do not. It would also be nice to have nightly snapshot builds to make it easier to test bug fixes.

**Comment by martijnlaan** [comment]
I agree but I don't have a server available to do these builds nor do I know of some free build service that supports Delphi.

**Comment by DeeHants** [comment]
Oh, I didn't get that this was CI builds of Inno itself. Feel free to reopen :)

**Comment by extensia** [comment]
Thanks for the feedback. I am glad to hear there is interest in continuous integration builds.

What are the OS requirements needed to setup the various Delphi compilers for a production Inno Setup build? I don't recall the exact OS requirements for the older versions of Delphi since it has been many years.

I have the ability to stand up virtual servers on our in-house VMware infrastructure and could setup a dedicated virtual machine to host the build process. Everything could be fully self-contained on a single VM and sandboxed on a dedicated VLAN accessible from the web.

Pulling code from Github on a scheduled or per-commit basis is trivial to configure with TeamCity or Jenkins. I personally prefer TeamCity since it is a bit more polished. Either way the license is free since Jenkins is open source and TeamCity is free for open source projects. The professional version of TeamCity is also free for commercial use up to 20 build configurations.

Thoughts?

**Comment by martijnlaan** [comment]
ANSI Inno Setup uses Delphi 2.01, Delphi 3.02 and Delphi 7. Unicode Inno Setup uses Delphi 2009 with Update 3. All of these work fine under Windows 8.1 as far as command line compilation is concerned which is the only part used by the build.

**Comment by extensia** [comment]
Glad to hear everything runs on modern versions of Windows. I was hoping we would not get stuck on XP since we just killed off our last XP VM a few months ago.

I have Delphi 2009, Delphi 7 and I believe a copy of Delphi 2 Desktop available. Not sure if it is Delphi 2.01. I definitely don't have Delphi 3.02 anymore since I moved to Delphi 5 early on due to some stability problems we had with the Delphi 3 compiler.

Do you know where I can find a copy of the Delphi 3 command line compiler? I owned a Desktop and Client Server license at one point, but it is way too old to show up in my EDN account. Not sure if any of the classic Borland tools sites are still on existence. They used to buy up old manuals and media as collectors items.

**Comment by martijnlaan** [comment]
Well, I have a copy of Delphi 3 myself obviously but otherwise I wouldn't know where to find it.
